### PR TITLE
Add new command to remove all users with a specific role from a thread

### PR DESCRIPTION
**Pull request by jkiller295**
## Description
Add a command to remove all users from a thread with a specific role
`/removeuserbyrole @rolename`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/moderateUsers/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      .setName('name')
+      .setDescription('Tag a role to remove the list of matching users')
+      .setRequired(true)
+  );
+
+export const removeUserByRole = async (
+  interaction: ChatInputCommandInteraction
+) => {
+  const channel = interaction.channel as ThreadChannel;
+  if (!channel?.isThread) return;
```
Comment (nqngo) [inline src/commands/moderateUsers/index.ts:24]: If not thread, interactions.reply("You can't remove all users with role from entire channel. This command only works in a thread.")

File: src/commands/moderateUsers/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    await interaction.reply(
+      "You don't have enough permission to run this command."
+    );
+    return;
+  }
+  const role = interaction.options.getRole('name', true);
+  const memberList = channel.members.cache.filter((user) =>
+    user.guildMember?.roles.cache.some((r) => r.id === role.id)
+  );
+  console.log(memberList.size);
```
Comment (nqngo) [inline src/commands/moderateUsers/index.ts:37]: Add timestamp so we can review the log.

Comment (nqngo) [reply src/commands/moderateUsers/index.ts:37]: Please also change commit message to something more meaningful.

Comment (samhwang) [reply src/commands/moderateUsers/index.ts:37]: cái này ổng đang test thôi, tới lúc merge vô sẽ xoá

File: src/commands/moderateUsers/index.ts (hunk 1 above)
Comment (jkiller295) [reply src/commands/moderateUsers/index.ts:24]: Can you technically remove users from a channel?

File: src/commands/moderateUsers/index.ts (hunk 2 above)
Comment (jkiller295) [reply src/commands/moderateUsers/index.ts:37]: > Add timestamp so we can review the log.

Where do we store this info?

File: src/commands/moderateUsers/index.ts (hunk 1 above)
Comment (samhwang) [reply src/commands/moderateUsers/index.ts:24]: technically you can't, but this ensures it doesn't arrive at that "bad" path

File: src/commands/moderateUsers/index.ts (hunk 2 above)
Comment (nqngo) [reply src/commands/moderateUsers/index.ts:37]: chạy docker thì console.log nó in ra mà. Chủ yếu có gì check nhanh 1 cái thì biết lúc nào nó chạy thôi chứ store làm gì

Comment (nqngo) [reply src/commands/moderateUsers/index.ts:37]: > cái này ổng đang test thôi, tới lúc merge vô sẽ xoá

Yeah, nhưng review thì vẫn phải comment. Done my parts. Lol.

File: src/commands/moderateUsers/indext.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { it, describe, expect } from 'vitest';
+import { removeUserByRole } from '.';
+import { isAdmin, isModerator } from '../../utils';
+import { Collection, Role, ThreadMember } from 'discord.js';
+
+vi.mock('../../utils/isSentFromAdmin');
+const mockIsSentFromAdmin = vi.mocked(isAdmin);
+vi.mock('../../utils/isSentFromModerator');
```
Comment (samhwang) [inline src/commands/moderateUsers/indext.test.ts:8]: This module doesn't exist.

**Review by samhwang — CHANGES_REQUESTED** [review]

**Comment by samhwang** [comment]
Please also update the ChangeLog

**Review by samhwang — APPROVED** [review]

**Pull request merged**